client/src/pages/locations/GeoLocation.js

```diff
@@ -113,25 +116,31 @@ const MGRSFormField = ({
   lat,
   lng,
   editable,
-  setValues,
+  setFieldValue,
   setFieldTouched,
   isSubmitting
 }) => {
   const [mgrs, setMgrs] = useState("")
 
-  useEffect(() => {
-    if (!editable && lat === null && lng === null) {
-      setMgrs("")
-    } else {
-      const mgrsValue = convertLatLngToMGRS(lat, lng)
-      if (mgrsValue) {
-        setMgrs(mgrsValue)
-        if (editable) {
-          setValues({ displayedCoordinate: mgrsValue, lat: lat, lng: lng })
+  // Coordinate should be sync between input field and map interaction
+  useEffect(
+    function syncMapEdit() {
```

Comment:
I would prefer an arrow function in here, for consistency with the rest of the code, I think all our code uses this form useEffect(() => {...}, [...])
Personally while reading this I first thought you are defining a function to use further in the effect so I thought this is not being used anywhere ...

Reply:
The problem is the function inside doesn't really give the purpose of why it exists. That is why i wanted to name it and for the stack tracing in error messages. We have a form field which can be edited from 2 places. 1- text input 2- Interactive map dragging the marker. Maybe comment is enough.